src/RulesEngine/RulesEngine/ExpressionBuilders/LambdaExpressionBuilder.cs

```diff
@@ -23,14 +23,17 @@ internal sealed class LambdaExpressionBuilder : RuleExpressionBuilderBase
         internal LambdaExpressionBuilder(ReSettings reSettings)
         {
             _reSettings = reSettings;
-        }
+        }
+
         internal override Expression<Func<RuleInput, RuleResultTree>> BuildExpressionForRule(Rule rule, IEnumerable<ParameterExpression> typeParamExpressions, ParameterExpression ruleInputExp)
         {
             try
             {
                 var config = new ParsingConfig { CustomTypeProvider = new CustomTypeProvider(_reSettings.CustomTypes) };
                 var e = DynamicExpressionParser.ParseLambda(config, typeParamExpressions.ToArray(), null, rule.Expression);
-                var body = (BinaryExpression)e.Body;
+                var body = e.Body is MethodCallExpression
+                            ? Expression.MakeBinary(ExpressionType.And, e.Body, Expression.Constant(true))
+                            : (BinaryExpression)e.Body;

```

Comment:
How about allowing all expressions? That will also allow unary expressions such as `!isSuccess`
```csharp
var body = e.Body is BinaryExpression
                            ? (BinaryExpression)e.Body
                            : Expression.MakeBinary(ExpressionType.And, e.Body, Expression.Constant(true))
```

Reply:
Seems like a good idea.
I'll take a look at the documentation for other possible expressions, possibly maintaining the same approach "behind the scenes" evaluating them to true, as is.

Reply:
Oh I didn't notice you change the code snippet. Yea that seems automatically better. Just need to check how the other expressions behave. Maybe by adding some test cases.

Reply:
@abbasc52 Done some testing and it works great! MemberAcessExpression/ UnaryExpression and MethodCallExpression which seem to be the only other expressions that will fall on your LambdaExpression parser will evaluate correctly as long as they produce a boolean value.

Added more tests.
I also remembered to modify the script so that it uses pattern matching in case the type check is successful, and therefore we avoid doing an explicit cast after the type check.

Reply:
Thanks for all the efforts 👍

Reply:
It seems your commit are not signed. So only admins can merge the PR. Shall i go ahead and merge it to master?

Reply:
Signed, but I don't think I can merge it still. If everything's okay, you may go ahead and merge it then.